cypress/e2e/room/room-header.spec.ts

```diff
@@ -0,0 +1,199 @@
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should render default buttons properly", () => {
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.get(".mx_RoomHeader").within(() => {
+            // Names (aria-label) of every button rendered on mx_RoomHeader by default
+            const expectedButtonNames = [
+                "Room options", // The room name button next to the room avatar, which renders dropdown menu on click
+                "Voice call",
+                "Video call",
+                "Search",
+                "Threads",
+                "Notifications",
+                "Room info",
+            ];
+
+            // Assert they are found and visible
+            for (const name of expectedButtonNames) {
+                cy.findByRole("button", { name }).should("be.visible");
+            }
+
+            // Assert that just those seven buttons exist on mx_RoomHeader by default
+            cy.findAllByRole("button").should("have.length", 7);
+        });
+
+        cy.get(".mx_RoomHeader").percySnapshotElement("Room header");
+    });
+
+    it("should render the pin button for pinned messages card", () => {
+        cy.enableLabsFeature("feature_pinning");
+
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.getComposer().type("Test message{enter}");
+
+        cy.get(".mx_EventTile_last").realHover().findByRole("button", { name: "Options" }).click();
+
+        cy.findByRole("menuitem", { name: "Pin" }).should("be.visible").click();
+
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findByRole("button", { name: "Pinned messages" }).should("be.visible");
+        });
+    });
+
+    it("should render a very long room name without collapsing the buttons", () => {
+        const LONG_ROOM_NAME =
+            "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore " +
+            "et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut " +
+            "aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum " +
+            "dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui " +
+            "officia deserunt mollit anim id est laborum.";
+
+        cy.createRoom({ name: LONG_ROOM_NAME }).viewRoomByName(LONG_ROOM_NAME);
+
+        cy.get(".mx_RoomHeader").within(() => {
+            // Wait until the room name is set
+            cy.get(".mx_RoomHeader_nametext").within(() => {
+                cy.findByText(LONG_ROOM_NAME).should("exist");
+            });
+
+            // Assert the size of buttons on RoomHeader are specified and the buttons are not compressed
+            // Note these assertions do not check the size of mx_RoomHeader_name button
+            // TODO: merge the assertions by using the same class name
```

Comment:
It will be achieved by https://github.com/matrix-org/matrix-react-sdk/pull/10495.